types.go

```diff
@@ -202,6 +202,12 @@ const (
 	PinByName
 )
 
+// LoadPinOptions control how a pinned object is loaded.
+type LoadPinOptions struct {
```

Comment:
Just thinking out loud:

Could this become a `LoadOptions` instead? `Flags uint32` is already quite generic and could be re-used across the public API surface to mean different things in different contexts.

Bitfields are difficult to use because they are a leaky abstraction (if you can call them an abstraction at all) and not very expressive or helpful to the user. If `LoadPinOption`'s only purpose is carrying rd/rw flags, what about the following:

```go
type LoadPinOptions struct {
  ReadOnly bool
  WriteOnly bool
}

func (opts LoadPinOptions) marshal() uint32 {
  var out uint32
  if opts.ReadOnly {
    out |= 1 << 0
  }
  if opts.WriteOnly {
    out |= 1 << 1
  }
  // .. etc ..
  return out
}
```
?

Reply:
Once we accept `LoadOptions` in more than one place it'll tie our hands as to what we can add to the struct. Basically, if we feed the type into two functions LoadOptions has to be the intersection of their two behaviours.

The booleans would be nicer, agreed. I'll play with it.

Reply:
I added your suggestion, which is an improvement overall I think. I kept `LoadPinOptions.Flags` because it's important to provide an escape hatch for users. Without it my "brilliant" test strategy with `math.MaxUint32` also doesn't work.

I've had to resort to a bit of a hack to expose `LoadPinOptions`: the type is defined in `internal`, and users get access to it via `type LoadPinOptions = internal.LoadPinOptions` which is kind of dodgy. This is done to avoid a circular dependency: I want to define `LoadPinOptions` in `ebpf` but don't want to make `LoadPinFlags` public.